src/main/java/com/legacyminecraft/poseidon/util/ServerLogRotator.java

```diff
@@ -0,0 +1,97 @@
+package com.legacyminecraft.poseidon.util;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.logging.*;
+
+/**
+ * The ServerLogRotator class handles the rotation of server logs.
+ * It archives the current log file and creates a new one every day at midnight.
+ */
+public class ServerLogRotator {
```

Comment:
bukkit has built-in scheduler semantics (see [here](https://github.com/RhysB/Project-Poseidon/tree/master/src/main/java/org/bukkit/scheduler)). I would probably use scheduleAsyncRepeatingTask so that the thread is managed by the pre-existing scheduler. It seems that this scheduler is used elsewhere in Project Poseidon to handle threading

Reply:
just note that the frequency is specified in Ticks so you would have to convert (i think 1 second is 20 ticks)

Reply:
I'll switch the logic to use the built in scheduler Bukkit provides!